### PR TITLE
New project antd

### PR DESCRIPTION
**Pull request by aminalakhsham**
Relates #22

### CODE REVIEW

File: client/src/components/Projects/ProjectNew/index.jsx [hunk 1]
```diff
@@ -1,9 +1,114 @@
...
-    </div>
-  )
+import React, { Component } from "react";
+import "./style.css";
+import TableMember from "./TableMember/TableMember";
+
+export default class index extends Component {
+  state = {
+    selection: {
+      rowKey: null,
```
Comment (ashatat) [inline client/src/components/Projects/ProjectNew/index.jsx]: I didn't understand why do you need the rowKey?
I don't see that you use it asside of the error checking

File: client/src/components/Projects/ProjectNew/index.jsx [hunk 2]
```diff
@@ -1,9 +1,114 @@
...
+              </div>
+              <div className="main-memberSelect">
+                <TableMember handleCheck={this.handleCheck} />
+              </div>
+            </div>
+            <div className="main-submit">
+              {this.state.error.errorStatus && (
+                <span className="error">{this.state.error.errorMsg}</span>
+              )}
+              <input className="main-add" type="submit" value="Add" />
```
Comment (ashatat) [inline client/src/components/Projects/ProjectNew/index.jsx]: the `value` should be `submit` or `create project`
`Add` here made me think it's just to add members :thinking:

File: client/src/components/Projects/index.jsx [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+export default class Projects extends Component {
+  render() {
+    return (
+      <div>
+        Projects
+        <Link to={"/project/1/1"}>project 1</Link>
+        <Link to={"/project/2/1"}>project 2</Link>
+      </div>
+    );
+  }
```
Comment (ashatat) [inline client/src/components/Projects/index.jsx]: FGS
why did you commit these changes :sob:

File: client/src/components/scrum/index.jsx [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { Link } from "react-router-dom";
 export default class Scrum extends Component {
   render() {
     const id = this.props.match.params.scrumId;
+    console.log(id)
```
Comment (ashatat) [inline client/src/components/scrum/index.jsx]: :crying_cat_face: 
seriously?
You commit `console.log`  :disappointed:

File: client/src/components/Layout/Header/index.jsx [hunk 5]
```diff
@@ -1,34 +1,52 @@
-import React, { Component } from 'react';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import './style.css';
+import React, { Component } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { NavLink } from "react-router-dom";
+
+import "./style.css";
```
[the remark this answers is not in the document]
Comment (aminalakhsham) [reply client/src/components/Layout/Header/index.jsx]: i think there is some error and you approve PR 
- No alt for img
- and use a tag with out href and should use NavLinke
but anyway i'll remove my change and make a new PR

File: client/src/components/Projects/ProjectNew/TableMember/member.json [hunk 6]
```diff
@@ -0,0 +1,32 @@
+[
+  {
+    "id": 1,
+    "name": "amin",
+    "role": "developer"
+  },
+  {
+    "id": 2,
+    "name": "amin",
```
[the remark this answers is not in the document]
Comment (aminalakhsham) [reply client/src/components/Projects/ProjectNew/TableMember/member.json]: i need the same to make the filter function

File: client/src/components/Projects/index.jsx (hunk 3 above)
Comment (aminalakhsham) [reply client/src/components/Projects/index.jsx]: remove change

File: client/src/components/scrum/index.jsx (hunk 4 above)
Comment (aminalakhsham) [reply client/src/components/scrum/index.jsx]: remove change

File: client/src/components/Projects/ProjectNew/TableMember/TableMember.jsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+      dataIndex: "role",
+      key: "role",
+      width: 150,
+      filters: this.state.filterArray[2],
+      onFilter: (value, record) => record.role.indexOf(value) === 0
+    }
+  ];
+
+  render() {
+    console.log(this.state.filterArray[0]);
```
Comment (IsraaSulaiman) [inline client/src/components/Projects/ProjectNew/TableMember/TableMember.jsx]: console!!!

File: client/src/components/Layout/Header/index.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (ashatat) [reply client/src/components/Layout/Header/index.jsx]: you are right it has some issues but it should be on a separate PR with a new issue
please remove it from this PR

File: client/src/components/Projects/ProjectNew/TableMember/TableMember.jsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+import "./style.css";
+import filterData from "./logic";
+
+export default class TableMember extends Component {
+  constructor(props) {
+    super(props);
+    this.state.member = require("./member.json");
+    this.state.filterArray = filterData(this.state.member);
+    this.columns[0].filters = this.state.filterArray[1];
+    this.columns[1].filters = this.state.filterArray[2];
```
Comment (ashatat) [inline client/src/components/Projects/ProjectNew/TableMember/TableMember.jsx]: **The constructor for a React component is called before it is mounted.**

so if your request to get the member take time your page will not load till it finished
which mean you can't implement loading.. indicator to tell the user that your page is fetching data from the server

**Also, ** `this.columns[0].filters` will not be modified if the state changed for some reason which means you have to refresh the page ("unmount then mount the component") to apply the changes to the filters

I'd advise fetching the data and updating the state (`this.setState({})) on `componentDidMount`
Also, assigning the filters inside the `render` function, so the filters will be updated when the `state` changes

File: client/src/components/Projects/index.jsx [hunk 9]
```diff
@@ -9,4 +9,4 @@ export default function Projects() {
       <Link to={'/project/2/1'}>project 2</Link>
     </div>
   );
-}
+}
```
Comment (ashatat) [inline client/src/components/Projects/index.jsx]: :thinking: 
why you commit this change?

File: client/src/components/scrum/index.jsx [hunk 10]
```diff
@@ -12,4 +13,4 @@ export default class Scrum extends Component {
       </>
     );
   }
-}
+}
```
Comment (ashatat) [inline client/src/components/scrum/index.jsx]: please, don't commit any changes outside your PR needs :pray:

File: client/src/components/Projects/index.jsx (hunk 9 above)
Comment (aminalakhsham) [reply client/src/components/Projects/index.jsx]: cuz remove my old change

File: client/src/components/scrum/index.jsx (hunk 10 above)
Comment (aminalakhsham) [reply client/src/components/scrum/index.jsx]: cuz remove my old change

File: client/src/components/Layout/Header/index.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (aminalakhsham) [reply client/src/components/Layout/Header/index.jsx]: i remove it

File: client/src/components/Projects/ProjectNew/index.jsx [hunk 11]
```diff
@@ -1,9 +1,111 @@
...
+      errorStatus: false,
+      errorMsg: ""
+    }
+  };
+  handleOnChange = e => {
+    const targetValue = e.target.value;
+    const inputName = e.target.name;
+    this.setState(prevState => {
+      const updatedState = { ...prevState.newProject };
+      updatedState[inputName] = targetValue;
```
Comment (AymanAlqoqa) [inline client/src/components/Projects/ProjectNew/index.jsx]: why you did not updated description!!!

**Pull request merged**